src/page/bank-payment.page.ts

```diff
@@ -4,6 +4,9 @@ export class BankPaymentPage {
   private confirmOrderButton: ElementFinder;
 
   constructor () {
+    /* Here the button doesn't have any id, title or name, so it is difficult to locate it
+    on a better way than the one already implemented. Anyway, I would recommend to the developper to
+    include some id to the button */
     this.confirmOrderButton = $('#cart_navigation > button > span');
```

Comment:
I was thinking that it is not a good idea to locate an element by his related text because it is likely to be changed at any moment for different reasons (for example, the page is being translated) so I avoided to use the content of elements as locators, but I can try it anyway.

Reply:
Yeah you're right, but It is a supposition, in real projects that kind of requirements are clear. I want to you explore more types of locators besides css selectors. CSS Selectors are not a golden hammer.

Reply:
Fixed using element(by.buttonText...). Interesting...